rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/activityList.spec.ts

```diff
@@ -0,0 +1,363 @@
+        .spyOn(wrapper.vm.rundeckContext.rundeckClient, "executionBulkDelete")
+        .mockImplementation(() => Promise.resolve({ allsuccessful: true }));
+      await wrapper.vm.$nextTick();
+      const reportRows = wrapper.findAll('[data-testid="report-row-item"]');
+      expect(reportRows.length).toBe(reports.length - 2); // Update expected length based on deletion
+      expect(executionBulkDeleteSpy).toHaveBeenCalledWith({ ids: [42] });
+    });
+
+    it("trigger bulk actions - search", async () => {
+      axiosMock.get.mockImplementation((url, data) => {
+        if (url.includes("eventsAjax")) {
+          if (data?.params.recentFilter.length > 0) {
+            return Promise.resolve({
+              data: {
+                total: 0,
+                offset: 0,
+                reports: [{ ...reports[0], node: "1/0/1" }],
+              },
+            });
+          }
+          return Promise.resolve({
+            data: {
+              total: reports.length,
+              offset: 0,
+              reports: reports,
+            },
+          });
+        }
+        if (url.includes("running")) {
+          return Promise.resolve({
+            data: {
+              executions: mockRunningExecutions,
+            },
+          });
+        }
+        return Promise.resolve({ data: {} });
+      });
+      const wrapper = await shallowMountActivityList();
+      await wrapper.vm.$nextTick();
+      const filterButton = wrapper.find(
+        '[data-testid="activity-list-filter-button"]',
+      );
+      await filterButton.trigger("click");
+      await wrapper.vm.$nextTick();
+      const activityFilter = wrapper.findComponent(ActivityFilter);
+      await activityFilter.vm.$emit("update:modelValue", {
+        recentFilter: "testJobId",
+      });
+      await flushPromises();
+      // Verify the results
+      const reportItems = wrapper.findAll('[data-testid="report-row-item"]');
+      expect(reportItems.length).toBe(1);
+    });
+  });
+
+  it("navigates to the execution page", async () => {
+    axiosMock.get.mockImplementation((url) => {
+      if (url.includes("execution")) {
+        return Promise.resolve({
+          data: {
+            executions: mockRunningExecutions,
+          },
+        });
+      }
+      if (url.includes("eventsAjax")) {
+        return Promise.resolve({
+          data: {
+            total: 20,
+            offset: 0,
+            reports: reports,
+          },
+        });
+      }
+      return Promise.resolve({ data: {} });
+    });
+    const wrapper = await shallowMountActivityList();
+    await wrapper.vm.$nextTick();
+    const reportRowItem = wrapper.find('[data-testid="report-row-item"]');
+    expect(reportRowItem.exists()).toBe(true);
+    await reportRowItem.trigger("click");
+    expect(window.location).toBe("/project/aaa/execution/show/42");
+  });
+
+  it("fetches data automatically and renders message when there are no executions since timestamp", async () => {
```

Comment:
This test's title is wrong, as it's asserting that there are 5 executions since the last check.